src/native/fs/fs_napi.cpp

```diff
@@ -284,66 +432,82 @@ struct FSWrapWorker : public FSWorker
 struct Stat : public FSWorker
 {
     std::string _path;
+    bool _use_lstat = false;
+    bool _skip_user_xattr = false;
     struct stat _stat_res;
+    XattrMap _xattr;
 
     Stat(const Napi::CallbackInfo& info)
         : FSWorker(info)
     {
         _path = info[1].As<Napi::String>();
+        if ((int)info.Length() == 3) {
+            Napi::Object options = info[2].As<Napi::Object>();
+            if (options.Has("use_lstat")) _use_lstat = options.Get("use_lstat").ToBoolean();
+            if (options.Has("skip_user_xattr")) _skip_user_xattr = options.Get("skip_user_xattr").ToBoolean();
+        }
         Begin(XSTR() << "Stat " << DVAL(_path));
     }
     virtual void Work()
     {
-        int r = stat(_path.c_str(), &_stat_res);
+        int flags = O_RDONLY;
+        // using O_PATH (MAC - O_SYMLINK) with O_NOFOLLOW allow us to open the symlink itself
+        // instead of openning the file it links to https://man7.org/linux/man-pages/man7/symlink.7.html
+#ifdef __APPLE__
+        if (_use_lstat) flags = O_SYMLINK | O_NOFOLLOW;
```

Comment:
did you intentionally override the flags (O_RDONLY)? or did you mean to add to it with `flags |= ...` ?

Reply:
yes, O_PATH is replacing O_RDONLY as is stronger than him, you can't even read the file with it.
```
perform operations that act purely at the file descriptor level.  The file itself is not opened, and
other file operations read, write ...  fail with the error EBADF.
```
When O_PATH is specified in flags, flag bits other than O_CLOEXEC, O_DIRECTORY, and O_NOFOLLOW are ignored.
```

Reply:
ok, worth noting it in a comment as it is not easy to notice